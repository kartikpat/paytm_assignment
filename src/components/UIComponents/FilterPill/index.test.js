import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilterPill from './index';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
    wrapper = shallow(<FilterPill label="Dummy Data" />);
});

describe('<FilterPill /> rendering', () => {
    it('should change the state selected when onClick function of the filter pill is invoked', () => {
        wrapper.props().onClick();
        expect(wrapper.props().selected).toEqual(true);
    });
});
