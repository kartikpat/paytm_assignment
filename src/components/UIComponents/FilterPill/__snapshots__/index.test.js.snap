// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Pill changes the class when clicked 1`] = `
<div
  className="sc-AxjAm iYFFrh"
  onClick={[Function]}
  selected={false}
>
  Dummy Data
</div>
`;

exports[`Filter Pill changes the class when clicked 2`] = `
<div
  className="sc-AxjAm iYFFrh"
  onClick={[Function]}
  selected={false}
>
  Dummy Data
</div>
`;

exports[`Filter Pill changes the class when clicked 3`] = `
<div
  className="sc-AxjAm iYFFrh"
  onClick={[Function]}
  selected={false}
>
  Dummy Data
</div>
`;
